threads/src/main/kotlin/pl/edu/pw/ia/threads/domain/query/projector/ThreadProjector.kt

```diff
@@ -52,6 +58,29 @@ class ThreadProjector(
 		repository.delete(event.threadId)
 	}
 
+	@EventHandler
+	fun on(event: AccountUpdatedEvent) {
+		repository.findByAccountId(event.accountId)
+			.map { it.copy(username = event.name) }
+			.subscribe {repository.save(it)}
+	}
+
+	@EventHandler
+	fun on(event: PostCreatedEvent) {
+		val view = repository.findByIdAndPostIdIsNull(event.threadId) ?: throw ThreadNotFoundException(event.threadId)
+		if(view.postId == event.postId){
+			repository.save(view.copy(postId = event.postId, post = event.content))
+		}
+	}
+
+	@EventHandler
+	fun on(event: PostUpdatedEvent) {
+		val threadView = repository.findByIdAndPostId(event.threadId, event.postId)
+		if(threadView != null && event.content != threadView.post) {
```

Comment:
ah, the famous string comparison error - in java we compare strings using equals
here you can use `!StringUtils.equals(event.content, threadView.post)`